src/cluster.cc

```diff
@@ -577,9 +580,14 @@ Status Cluster::ParseClusterNodes(const std::string &nodes_str, ClusterNodes *no
           }
         }
       } else if (ranges.size() == 2) {
-        start = std::atoi(ranges[0].c_str());
-        stop = std::atoi(ranges[1].c_str());
-        if (start >= stop || start < 0 || stop >= kClusterSlots) {
+        auto parse_start = ParseInt<int>(ranges[0].c_str(), NumericRange<int>{0, kClusterSlots - 1}, 10);
+        auto parse_stop = ParseInt<int>(ranges[1].c_str(), NumericRange<int>{0, kClusterSlots - 1}, 10);
+        if (!parse_start || !parse_stop) {
+          return Status(Status::ClusterInvalidInfo, "Invalid cluste slot range");
+        }
+        start = *parse_start;
+        stop = *parse_stop;
+        if (start >= stop) {
           return Status(Status::ClusterInvalidInfo, "Invalid cluste slot range");
         }
```

Comment:
```suggestion
        if (!parse_start || !parse_stop || *parse_start >= *parse_stop) {
          return Status(Status::ClusterInvalidInfo, "Invalid cluste slot range");
        }
        start = *parse_start;
        stop = *parse_stop;
```

Reply:
Hi. Although your modification is logically correct, a warning will be issued. That's what I wrote before, because the warning changed to what it is now.

Reply:
Thanks, great advice.